metadata/src/main/java/org/apache/kafka/controller/ConfigurationControlManager.java

```diff
@@ -308,6 +330,48 @@ private ApiError validateAlterConfig(ConfigResource configResource,
         return ApiError.NONE;
     }
 
+    void maybeTriggerPartitionUpdateOnMinIsrChange(List<ApiMessageAndVersion> records) {
```

Comment:
I guess eventually we will probably want some more general system for responding to configuration changes, and not special-case min.isr. However, we don't have to do that in this PR.

I also understand why you had this function receive a list of records here rather than something fancier (easier to integrate into the existing code, and into the two configuration change paths.)